spaceR/spaceR/ViewModels/SignupViewModel.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  SignupViewModel.swift
+//  SpaceR
+//
+//  Created by Kieran Woodrow on 2022/03/13.
+//
+
+import Foundation
+import UIKit
+
+class SignupViewModel {
+    
+    private var userFirstName = ""
+    private var userLastName = ""
+    private var userEmail = ""
+    private var userPassword = ""
+    // swiftlint:disable force_cast
+    private let coreDataPersistantObject = (UIApplication.shared.delegate as! AppDelegate).persistentContainer.viewContext
+    // swiftlint:enable force_cast
+    
+    func setUserFirstName(firstNameFromForm: String) {
```

Comment:
Naming improvement: `func set(firstName: String)`

Reply:
Can i not rather change the variable name and keep the function name. It makes more sense to me for a function to be explicit in what it's doing. e.g. func setFirstName(name: String)
?

Reply:
So the parameter name is part of the function name 😄 ideally we nearly want this function to read like a concise sentence.

so if we look at your original name: 
`func setUserFirstName(firstNameFromForm: String)`

If we were to call this function there would be too much redundancy:

`setUserFirstName(firstNameFromForm: "")`

We are referencing first name twice as well as we bring form into the equation so it is not very clear what this function actually does, what is form? a custom component? the ViewController? 

Whereas if you used something like: 

`func set(firstName: String) {`

Other developers would be quite sure that this function sets the first name of the user:

`set(firstName: "")`

You could also do something like this:

`func setFirstName(_ name: String) `
`setFirstName("")`

Let me know if you want to chat about this over a call 😄